core/drivers/crypto/caam/hal/common/hal_cfg.c

```diff
@@ -0,0 +1,228 @@
+		HAL_TRACE("CAAM device not defined or not enabled\n");
+		goto exit_get_conf;
+	}
+
+	jr_offset = dt_node_offset_by_compatible_status(fdt,
+			DT_STATUS_OK_SEC, &node, dt_jr_match_table);
+	if (jr_offset == 0) {
+		EMSG("No Job Ring defined in DTB\n");
+		goto exit_get_conf;
+	}
+
+	/* We took one job ring, make it unavailable for Normal World */
+	if (dt_disable_status(fdt, node)) {
+		EMSG("Not able to disable JR DTB entry\n");
+		goto exit_get_conf;
+	}
+
+	jr_cfg->base    = ctrl_base;
+	jr_cfg->offset  = jr_offset;
+	jr_cfg->nb_jobs = NB_JOBS_QUEUE;
+
+	// Get the job ring interrupt
+	jr_cfg->it_num  = dt_get_irq(fdt, node);
+	if (jr_cfg->it_num < 0) {
+		EMSG("Job Ring interrupt number not defined in DTB\n");
+		goto exit_get_conf;
+	}
+	// Add index of the first SPI interrupt
+	jr_cfg->it_num += 32;
+#else
+
+	ctrl_base = (vaddr_t)phys_to_virt(CAAM_BASE, MEM_AREA_IO_SEC);
+	if (!ctrl_base) {
+		if (!core_mmu_add_mapping(MEM_AREA_IO_SEC, CAAM_BASE,
+			    CORE_MMU_PGDIR_SIZE)) {
+			EMSG("Unable to map CAAM Registers");
+			goto exit_get_conf;
+		}
+
+		ctrl_base = (vaddr_t)phys_to_virt(CAAM_BASE, MEM_AREA_IO_SEC);
+	}
+
+	if (!ctrl_base) {
+		EMSG("Unable to get the CAAM Base address");
+		goto exit_get_conf;
+	}
+
+	jr_cfg->base    = ctrl_base;
+	jr_cfg->offset  = (CFG_JR_INDEX + 1) * JRx_BLOCK_SIZE;
+	jr_cfg->nb_jobs = NB_JOBS_QUEUE;
+	// Add index of the first SPI interrupt
+	jr_cfg->it_num  = CFG_JR_IRQ;
+#endif // CFG_CAAM_DT
+
+	retstatus = CAAM_NO_ERROR;
+
+exit_get_conf:
+	HAL_TRACE("HAL CFG Get Job Ring returned (0x%x)\n", retstatus);
+	return retstatus;
+}
+
+/**
+ * @brief   Setup the Non-Secure Job Ring
+ *
+ * @param[in] ctrl_base   Virtual CAAM Controller Base address
+ *
+ */
+void hal_cfg_setup_nsjobring(vaddr_t ctrl_base)
+{
+	enum CAAM_Status status __maybe_unused;
+
+	paddr_t jr_offset;
+
+#ifdef CFG_CAAM_DT
+
+	void *fdt;
+	int  node;
+
+	fdt = get_dt();
+	if (!fdt) {
+		EMSG("DTB no present\n");
+		return;
+	}
+
+	node = fdt_node_offset_by_compatible(fdt, 0, dt_ctrl_match_table);
+
+	if (node < 0) {
+		EMSG("Caam Node not found err = 0x%X\n", node);
+		return;
+	}
+
+	/* Configure the other Job ring to be Non-Secure */
+	do {
+		jr_offset = dt_node_offset_by_compatible_status(fdt,
+				DT_STATUS_OK_NSEC, &node, dt_jr_match_table);
+		if (jr_offset != 0) {
+			status = hal_jr_setowner(ctrl_base, jr_offset,
+					JROWN_ARM_NS);
+			HAL_TRACE("JR setowner returned %x", status);
+			if (status == CAAM_NO_ERROR)
+				hal_jr_prepare_backup(ctrl_base, jr_offset);
+		}
+	} while (jr_offset != 0);
+
+#else
+
+	uint8_t jrnum;
+
+	jrnum = hal_ctrl_jrnum(ctrl_base);
+
+	/* Configure the other Job ring to be Non-Secure */
+	do {
+#ifdef CFG_CRYPTO_DRIVER
+		if (jrnum != (CFG_JR_INDEX + 1)) {
+			jr_offset = jrnum * JRx_BLOCK_SIZE;
+			status = hal_jr_setowner(ctrl_base, jr_offset,
+				JROWN_ARM_NS);
+			if (status == CAAM_NO_ERROR)
+				hal_jr_prepare_backup(ctrl_base, jr_offset);
+		}
+#else
```

Comment:
Diff with and without  `CFG_CRYPTO_DRIVER` is not obvious. why skipped only when `CFG_CRYPTO_DRIVER!=y`
and why `jrnum == 0` is also skipped?
Suggest to replace this block with:
```
	do {
#ifdef CFG_CRYPTO_DRIVER
		/* some comment stating why this specific case */
		if (jrnum == CFG_JR_INDEX + 1)
			continue;
#endif
		jr_offset = jrnum * JRx_BLOCK_SIZE;
		status = hal_jr_setowner(ctrl_base, jr_offset, JROWN_ARM_NS);
		if (status == CAAM_NO_ERROR)
			hal_jr_prepare_backup(ctrl_base, jr_offset);
	} while (--jrnum);